runtime/ts/storage/pouchdb/pouch-db-collection.ts

```diff
@@ -0,0 +1,364 @@
+   * @param keys The CRDT keys used to store this object
+   * @param orginatorId TBD passed to event listeners
+   */
+  async store(value, keys: string[], originatorId=null) : Promise<void> {
+    assert(keys != null && keys.length > 0, 'keys required');
+    const id = value.id;
+    
+    const changeEvent = {value, keys, effective: undefined};
+
+    if (this.referenceMode) {
+      const referredType = this.type.primitiveType();
+      const storageKey = this.storageEngine.baseStorageKey(referredType, this.storageKey);
+
+      // Update the referred data
+      await this.getModelAndUpdate((crdtmodel) => {
+        changeEvent.effective = crdtmodel.add(value.id, {id: value.id, storageKey}, keys);
+        return crdtmodel;
+      });
+
+      await this.ensureBackingStore();
+      await this.backingStore.store(value, keys);
+    } else {
+      await this.getModelAndUpdate((crdtmodel) => {
+        // check for existing keys?
+        changeEvent.effective = crdtmodel.add(value.id, value, keys);
+        return crdtmodel;
+      });
+    }
+
+    this.version++;
+
+    // 2. Notify Listeners
+    this._fire('change', {add: [changeEvent], version: this.version, originatorId});
+  }
+
+
+  /**
+   * Remove ids from a collection for specific keys.
+   * @param id the id to remove.
+   * @param keys the CRDT specific keys to remove.
+   * @param originatorId TBD passed to event listeners.
+   */
+  async remove(id, keys:string[] = [], originatorId=null) {
+
+    await this.getModelAndUpdate((crdtmodel) => {
+      if (keys.length === 0) {
+        keys = crdtmodel.getKeys(id);
+      }
+      const value = crdtmodel.getValue(id);
+
+      if (value !== null) {
+        const effective = crdtmodel.remove(id, keys);
+        // TODO(lindner): isolate side effects...
+        this.version++;
+        this._fire('change', {remove: [{value, keys, effective}], version: this.version, originatorId});
+      }
+      return crdtmodel;
+    });
+  }
+
+  /**
+   * Triggered when the storage key has been modified.  For now we
+   * just refetch and trigger listeners.  This is fast since the data
+   * is synced locally.
+   */
+  public onRemoteStateSynced() {
+    // updates internal state
+    this.getModel();
+  }
+
+  /**
+   * Updates the local model cache from PouchDB and returns the CRDT
+   * model for use.
+   */
+  private async getModel(): Promise<CrdtCollectionModel> {
+    try {
+      const result = await this.db.get(this.pouchDbKey.location);
+      
+      // compare revisions
+      if (this._rev !== result._rev) {
+        // remote revision is different, update local copy.
+        this._model = new CrdtCollectionModel(result['model']);
+        this._rev = result._rev;
+        this.version++;  // yuck.
+        // TODO(lindner): fire change events here?
+      }
+
+    } catch (err) {
+      this._model = new CrdtCollectionModel();
+      this._rev = undefined;
+      // TODO(lindner): throw
```

Comment:
Maybe at least log the error for now if we don't add the throw?

Reply:
switched to only reset on not_found, and now throws for other errors.